packages/perseus/src/widgets/radio/widget.jsx

```diff
@@ -443,28 +476,35 @@ class Radio extends React.Component<Props> {
             // Support legacy choiceState implementation
         }
         /* istanbul ignore if - props.values is deprecated */
-        // $FlowFixMe[prop-missing]
         if (props.values) {
             let noneOfTheAboveIndex = null;
             let noneOfTheAboveSelected = false;
 
-            const values = props.values.slice();
+            const choicesSelected = props.values.map(Boolean);
             const countChoices = props.countChoices;
             const numCorrect = props.numCorrect;
+            const valuesLength = props.values?.length ?? 0;
```

Comment:
Flow has this interesting behavior where it will invalidate type checks after function calls.  This is why even though we check if `props.values` is defined on line 479, we end up having to the same check on line 486.  The reason Flow does this, is that it possible that any function could mutate an object and Flow can't rule out that possibility.  The way around this is to store `props.values` in a new variable `values` and type check that variable and use it.  Here's what that would look like:
```
const {values} = props;
if (values) {
    let noneOfTheAboveIndex = null;
    let noneOfTheAboveSelected = false;

    const choicesSelected = [...values];
    const countChoices = props.countChoices;
    const numCorrect = props.numCorrect;
    const valuesLength = values.length;

    for (let i = 0; i < valuesLength; i++) {
        const index = props.choices[i].originalIndex;
        choicesSelected[index] = values[i];

        if (props.choices[i].isNoneOfTheAbove) {
            noneOfTheAboveIndex = index;
            if (choicesSelected[i]) {
                noneOfTheAboveSelected = true;
            }
        }
    }
    return {
        choicesSelected,
        noneOfTheAboveIndex,
        noneOfTheAboveSelected,
        countChoices,
        numCorrect,
    };
}
```
This results in the code having fewer type checks.  I've taken the liberty to replace the original `props.values.slice()` with `[...values]`.  Both make a copy of `values`.

Reply:
Thanks so much! Incorporated your suggested changes :D